src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs

```diff
@@ -69,7 +69,7 @@ internal sealed class InProcRemoteHostClient : RemoteHostClient
             // this is what consumer actually use to communicate information
             var serviceStream = await _inprocServices.RequestServiceAsync(serviceName, cancellationToken).ConfigureAwait(false);
```

Comment:
here's the other place where we also get two streams in a way that does not seem exception/dispose safe.  i think the core pattern of getting hte streams and making the optional snapshot, and then making the jsonrpcsession would ideally be in one place.  

It does mean passing along 3 funcs afaict, but it means that we can have one place that is written very carefully for resiliency.